pkgs/by-name/en/en-croissant/package.nix

```diff
@@ -0,0 +1,110 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, stdenvNoCC
+, jq
+, moreutils
+, nodePackages
+, cacert
+, rustPlatform
+, cargo
+, rustc
+, cargo-tauri
+, pkg-config
+, openssl
+, libsoup
+, wrapGAppsHook
+, gtk3
+, webkitgtk
+}:
+
+stdenv.mkDerivation rec {
+  pname = "en-croissant";
+  version = "0.8.0";
+
+  src = fetchFromGitHub {
+    owner = "franciscoBSalgueiro";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-imM7sPV2Yfmi2N/YqOhbiVRBGE/bzuIaqHI0aBHLmTU=";
+  };
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  pnpm-deps = stdenvNoCC.mkDerivation {
+    pname = "${pname}-pnpm-deps";
+    inherit src version;
+
+    nativeBuildInputs = [
+      jq
+      moreutils
+      nodePackages.pnpm
+      cacert
+    ];
+
+    installPhase = ''
+      export HOME=$(mktemp -d)
+      pnpm config set store-dir $out
+      # use --frozen-lockfile to avoid checking git deps
+      pnpm install --frozen-lockfile
+
+      # Remove timestamp and sort the json files
+      rm -rf $out/v3/tmp
+      for f in $(find $out -name "*.json"); do
+        sed -i -E -e 's/"checkedAt":[0-9]+,//g' $f
+        jq --sort-keys . $f | sponge $f
+      done
+    '';
+
+    dontFixup = true;
+    outputHashMode = "recursive";
+    outputHash = "sha256-DTsT30o2BcXqqQW7uKtIexGoPT1KTxqPgDxOh+16olY=";
```

Comment:
you should use something like `fetchNpmDeps` then you wont need to deal with fixed output derivations yourself

Reply:
Hi @kirillrdy, we need to create a `fetchPnpmDeps`. Unfortunately, this project uses only pnpm as its node-modules manager. The nix-packages with pnpm as node manager will be a little bit hard to read/maintain with this necessary workaround for downloading node-modules.

Another examples with the same implementation logic:

pkgs/applications/misc/pot/default.nix
pkgs/applications/networking/geph/default.nix
pkgs/by-name/ve/vesktop/package.nix

Reply:
ok i see

as you can see from ofborg, the hash is different. The difficulty with fixed output derivations is that they must always produce same output. I get same hash as ofborg ( this is on x86_64-linux ) we'll need to figure out why content is different and try to fix it.

Reply:
> ok i see
> 
> as you can see from ofborg, the hash is different. The difficulty with fixed output derivations is that they must always produce same output. I get same hash as ofborg ( this is on x86_64-linux ) we'll need to figure out why content is different and try to fix it.

Hmmmm, I believe that this problem are from the optional dependencies, like rollup and esbuild. I will work on a way to use the `nodePackages.<package-name>` in the build, and in this way we can ignore the download of optional dependencies with pnpm.